kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -655,14 +656,20 @@ internal class ObjCExportTranslatorImpl(
                     MethodBridgeValueParameter.ErrorOutParameter ->
                         ObjCPointerType(ObjCNullableReferenceType(ObjCClassType("NSError")), nullable = true)
 
-                    MethodBridgeValueParameter.SuspendCompletion -> {
+                    is MethodBridgeValueParameter.SuspendCompletion -> {
                         val resultType = when (val it = mapReferenceType(method.returnType!!, objCExportScope)) {
-                            is ObjCNonNullReferenceType -> ObjCNullableReferenceType(it, isNullableResult = false)
+                            is ObjCNonNullReferenceType -> {
+                                if (bridge.useUnitCompletion) {
```

Comment:
If `it is ObjCNullableReferenceType`, then the method should be exported as before, like `(KtKotlinUnit * _Nullable_result, NSError * _Nullable)` unless I'm missing something.

Reply:
Or do you mean that if `useUnitCompletion == true` we shouldn't actually bother checking what `it` is?

Reply:
> Or do you mean that if `useUnitCompletion == true` we shouldn't actually bother checking what `it` is?

Exactly.

> If `it is ObjCNullableReferenceType`, then the method should be exported as before, like `(KtKotlinUnit * _Nullable_result, NSError * _Nullable)` regardless of the feature flag state, unless I'm missing something.

Well, it depends. If `MethodBridge` says `useUnitCompletion`, then the generated code will expect single-argument completion. So if this code decides to show two-argument completion in the signature, then it will be inconsistent.
And again, I guess this shouldn't be possible, but better to be safe than sorry.

Reply:
I appreciate the elaboration, addressed.